packages/core/admin/admin/src/content-manager/components/DynamicZone/index.js

```diff
@@ -37,11 +37,33 @@ const DynamicZone = ({
 }) => {
   const toggleNotification = useNotification();
   const [isOpen, setIsOpen] = useState(false);
+  const [shouldOpenAddedComponent, setShouldOpenAddedComponent] = useState(false);
   const dynamicDisplayedComponentsLength = dynamicDisplayedComponents.length;
+
   const [componentCollapses, setComponentsCollapses] = useState(
     createCollapses(dynamicDisplayedComponentsLength)
   );
 
+  useEffect(() => {
+    setComponentsCollapses(createCollapses(dynamicDisplayedComponentsLength));
+  }, [dynamicDisplayedComponentsLength]);
+
+  useEffect(() => {
+    if (shouldOpenAddedComponent) {
+      setComponentsCollapses(prev =>
+        prev.map((collapse, index) => {
+          if (index === prev.length - 1) {
+            return { isOpen: true };
```

Comment:
Can you spread the collapse object before. If ever another value was added, it would be lost here. WDYT?

`{ ...collapse, isOpen: true }`

Reply:
The collapse is  an object that only has one key : `isOpen`

Reply:
I know but if we add another key to that object (for any reason), it will be lost here anyway.